src/System.Threading.Thread/tests/ThreadTests.cs

```diff
@@ -1025,5 +1025,49 @@ public static void MiscellaneousTest()
             Thread.SpinWait(1);
             Thread.Yield();
         }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public static void WindowsPrincipalPolicyTest_Windows()
+        {
+            DummyClass.RemoteInvoke(() =>
+            {
+                if (!string.IsNullOrEmpty(Environment.UserDomainName))
+                {
+                    AppDomain.CurrentDomain.SetPrincipalPolicy(PrincipalPolicy.WindowsPrincipal);
+                    Assert.Equal(Environment.UserDomainName + @"\" + Environment.UserName, Thread.CurrentPrincipal.Identity.Name);
+                }
+            }).Dispose();
+        }
+
+        [Fact]
+        public static void UnauthenticatedPrincipalTest()
+        {
+            DummyClass.RemoteInvoke(() =>
+            {
+                AppDomain.CurrentDomain.SetPrincipalPolicy(PrincipalPolicy.UnauthenticatedPrincipal);
+                Assert.Equal(string.Empty, Thread.CurrentPrincipal.Identity.Name);
+            }).Dispose();
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
```

Comment:
For these skips, it'd be nice to have a message in the attribute that states briefly why it's being skipped.